docs/content/docs/build/container-states-events.md

```diff
@@ -0,0 +1,336 @@
+
+Once it is deleted, the container cannot move back to the **published** state again.
+It is possible for a client to create a new container using the same schema as a deleted container. 
+If a record of the final data values of the deleted container are available to the creating client, they can be used to initialize the new container. 
+But the new container has a different ID from the deleted one and subsequent clients will have to connect to it using the new ID.
+
+#### Handling publication status
+
+Your code can test for the publication status with the [container.AttachState]({{< relref "/docs/apis/fluid-static/fluidcontainer-class.md#attachstate-property" >}}) property which has an [AttachState]({{< relref "/docs/apis/container-definitions.md#attachstate-enum" >}}) value. 
+This can be useful if your application will publish the container in some code paths, but not others. 
+For example, in complex cases, you may need to determine whether the container has already been published before you call `container.attach`. The following is a simple example.
+
+```typescript
+if (container.attachState !== AttachState.Attached) {
+    container.attach();
+}
+```
+
+How you handle the **publishing** (`AttachState.Attaching`) state depends on the situation. If you want to prevent `container.attach` from being called when it has already been called, then treat publishing (`Attaching`) the same as published (`Attached`). 
+On the other hand, if you want to block editing when until the container is fully published, treat publishing (`Attaching`) the same as unpublished (`Detached`).
+
+### Synchronization status states
+
+Synchronization status refers to whether the container's data on the client is saved to the Fluid service. It is a *client-relative state*: the container may have a different synchronization state on different clients. The following diagram shows the possible Synchronization states and the events that cause a state transition. Details are below the diagram.
+
+<!-- TO MODIFY THIS DIAGRAM, SEE INSTRUCTIONS AT THE BOTTOM OF THIS FILE. -->
+![A state diagram of the two possible Synchronization states](./images/SynchronizationStates.png)
+
+{{< callout note >}}
+
+The dotted arrow represents a boolean guard condition. It means that the container cannot transition out of the **dirty** state unless the container is published and connected.
+
+{{< /callout >}}
+
+- **dirty**: A container is in **dirty** state on a client in two situations:
+
+    - It is unpublished. 
+    - Changes have been made to the data on the client but not yet sent to the Fluid service, or sent but not yet acknowledged by the service.
+
+- **saved**: A container is in **saved** state on a client when the container has been published and the service has acknowledged all data changes made on the client. 
+
+When a new change is made to the data, the container moves to **dirty** state. When all pending changes are acknowledged, it transitions to the **saved** state. 
```

Comment:
```suggestion
When a new change is made to the data in a given client, the container moves to **dirty** state *in that client*. When all pending changes are acknowledged, it transitions to the **saved** state. 
```
Just to be clear that changes in other clients don't affect the local client's dirty/saved state.